test/test_treetensornetworks/test_solvers/test_dmrg.jl

```diff
@@ -41,6 +37,38 @@ using Test
     H, psi; nsweeps, maxdim, cutoff, nsite, solver_krylovdim=3, solver_maxiter=1
   )
   @test inner(psi', H, psi) ≈ inner(psi2', H_mpo, psi2)
+
```

Comment:
Note the new ability below to pass in a custom sweeping pattern from outside the algorithm, represented as a simple array of arrays (or array of tuples is ok).